### PR TITLE
proposer settings log ux

### PR DESCRIPTION
**Pull request by james-prysm**
**What type of PR is this?**

 Other

**What does this PR do? Why is it needed?**

Improves the logs to better handle debugging

**Which issues(s) does this PR fix?**

Fixes #

**Other notes for review**

### CODE REVIEW

File: config/proposer/loader/loader.go [hunk 1]
```diff
@@ -128,12 +129,22 @@ func (psl *settingsLoader) Load(cliCtx *cli.Context) (*proposer.Settings, error)
...
+			return string(b)
+		}())
 	}
 
 	// start to process based on load method
 	for _, method := range psl.loadMethods {
 		switch method {
 		case defaultFlag:
+			if psl.existsInDB {
+				log.Warn("Previously saved proposer settings were loaded from the DB, only default settings will be updated. Please provide new proposer settings or clear DB to reset proposer settings.")
```
Comment (nalepae) [inline config/proposer/loader/loader.go:146]: If I start the VC with a fresh DB:

```
bazel run //cmd/validator -- --holesky --suggested-fee-recipient 0x3A8734b7D0a66b0a99218a8dAd27933e50A65efB --validators-external-signer-url http://localhost:9000 --proposer-settings-file  /Users/manu/Downloads/proposer_settings.yaml
```

==> OK

If I restart the VC with the same command:
```
[2024-06-13 11:39:02]  WARN Previously saved proposer settings were loaded from the DB, only default settings will be updated. Please provide new proposer settings or clear DB to reset proposer settings.
```

By `default settings` are you talking about `defaultconfig` (by opposition to `proposerconfig`?
If yes, why do only the `default settings` will be updated in such a case?

**Review by nalepae — COMMENTED** [review]
Is is possible to use `withField` instead of `Infof`?

File: config/proposer/loader/loader.go [hunk 2]
```diff
@@ -128,12 +129,22 @@ func (psl *settingsLoader) Load(cliCtx *cli.Context) (*proposer.Settings, error)
 			return nil, err
 		}
 		loadConfig = dbps.ToConsensus()
+		log.Debugf("DB Loaded Proposer Settings: %s", func() string {
```
Comment (rkapka) [inline config/proposer/loader/loader.go:132]:
```suggestion
		log.Debugf("DB loaded proposer settings: %s", func() string {
```

File: config/proposer/loader/loader.go (hunk 1 above)
Comment (james-prysm) [reply config/proposer/loader/loader.go:146]: hmm you're right it doesn't i should probably change this log

Comment (james-prysm) [reply config/proposer/loader/loader.go:146]: if you provide the proposer settings file flag it should also be updated.

**Review by terencechain — APPROVED** [review]

**Review by nalepae — APPROVED** [review]

**Pull request merged**